src/core/config/openthread-core-config-check.h

```diff
@@ -546,4 +546,13 @@
        "(and OPENTHREAD_CONFIG_LOG_DEFINE_AS_MACRO_ONLY)"
 #endif
 
+#if OPENTHREAD_CONFIG_MESSAGE_USE_HEAP_ENABLE
+#if !OPENTHREAD_CONFIG_DTLS_ENABLE
+#warning "OPENTHREAD_CONFIG_MESSAGE_USE_HEAP_ENABLE is strongly discouraged when OPENTHREAD_CONFIG_DTLS_ENABLE is off."
```

Comment:
👍

Reply:
I am afraid that the warning might not be noticed by users. 
Since we haven't set proper heap size for each platform when DTLS is OFF, maybe it's safer to prohibit using `MESSAGE_USE_HEAP` in such case?

Not to mention, using `MESSAGE_USE_HEAP` when DTLS is OFF is basically not helpful.

Reply:
I agree. `#error` here makes more sense to me as well.

Reply:
@bukepo , thoughts?

Reply:
```suggestion
#error "OPENTHREAD_CONFIG_MESSAGE_USE_HEAP_ENABLE is strongly discouraged when OPENTHREAD_CONFIG_DTLS_ENABLE is off."
```